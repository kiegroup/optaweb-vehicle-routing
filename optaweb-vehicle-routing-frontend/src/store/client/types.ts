import { Viewport as LeafletViewport } from 'react-leaflet';
import { Action } from 'redux';

export enum ActionType {
  UPDATE_VIEWPORT = 'UPDATE_VIEWPORT',
  RESET_VIEWPORT = 'RESET_VIEWPORT',
}

export interface UpdateViewportAction extends Action<ActionType.UPDATE_VIEWPORT> {
  value: LeafletViewport;
}

export interface ResetViewportAction extends Action<ActionType.RESET_VIEWPORT> {
}

export interface UserViewport {
  isDirty: boolean;
  center: [number, number];
  zoom: number;
}

export type ViewportAction =
  | UpdateViewportAction
  | ResetViewportAction;
