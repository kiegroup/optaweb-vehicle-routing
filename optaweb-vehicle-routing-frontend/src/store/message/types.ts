import { Action } from 'redux';

export enum ActionType {
  RECEIVE_MESSAGE = 'RECEIVE_MESSAGE',
  READ_MESSAGE = 'READ_MESSAGE',
}

export interface ReceiveMessageAction extends Action<ActionType.RECEIVE_MESSAGE> {
  readonly payload: MessagePayload;
}

export interface ReadMessageAction extends Action<ActionType.READ_MESSAGE> {
  readonly id: string;
}

export type MessageAction = ReceiveMessageAction | ReadMessageAction;

export interface MessagePayload {
  readonly id: string;
  readonly text: string;
}

export interface Message extends MessagePayload {
  readonly status: 'new' | 'read';
}
