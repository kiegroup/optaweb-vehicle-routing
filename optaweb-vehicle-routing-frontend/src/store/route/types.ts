/*
 * Copyright 2018 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Action } from 'redux';

export interface LatLng {
  readonly lat: number;
  readonly lng: number;
}

export interface LatLngWithDescription extends LatLng {
  description: string;
}

export interface Location extends LatLng {
  readonly id: number;
  // TODO decide between optional, nullable and more complex structure (displayName, fullDescription, address, ...)
  readonly description?: string;
}

export interface Vehicle {
  readonly id: number;
  readonly name: string;
  readonly capacity: number;
}

export interface Route {
  readonly vehicle: Vehicle; // TODO change to vehicleId
  readonly visits: Location[];
}

export type LatLngTuple = [number, number];

export interface RouteWithTrack extends Route {
  readonly track: LatLngTuple[];
}

export interface RoutingPlan {
  readonly distance: string;
  readonly vehicles: Vehicle[];
  readonly depot: Location | null;
  readonly visits: Location[];
  readonly routes: RouteWithTrack[];
}

export enum ActionType {
  UPDATE_ROUTING_PLAN = 'UPDATE_ROUTING_PLAN',
  DELETE_LOCATION = 'DELETE_LOCATION',
  ADD_LOCATION = 'ADD_LOCATION',
  ADD_VEHICLE = 'ADD_VEHICLE',
  DELETE_VEHICLE = 'DELETE_VEHICLE',
  CLEAR_SOLUTION = 'CLEAR_SOLUTION',
}

export interface AddLocationAction extends Action<ActionType.ADD_LOCATION> {
  readonly value: LatLngWithDescription;
}

export interface AddVehicleAction extends Action<ActionType.ADD_VEHICLE> {
}

export interface ClearRouteAction extends Action<ActionType.CLEAR_SOLUTION> {
}

export interface DeleteLocationAction extends Action<ActionType.DELETE_LOCATION> {
  readonly value: number;
}

export interface DeleteVehicleAction extends Action<ActionType.DELETE_VEHICLE> {
  readonly value: number;
}

export interface VehicleCapacity {
  vehicleId: number;
  capacity: number;
}

export interface UpdateRouteAction extends Action<ActionType.UPDATE_ROUTING_PLAN> {
  readonly plan: RoutingPlan;
}

export type RouteAction =
  | AddLocationAction
  | AddVehicleAction
  | DeleteLocationAction
  | DeleteVehicleAction
  | UpdateRouteAction
  | ClearRouteAction;
