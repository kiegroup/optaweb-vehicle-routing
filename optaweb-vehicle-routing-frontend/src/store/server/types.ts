import { Action } from 'redux';
import { LatLng } from '../route/types';

export enum ActionType {
  SERVER_INFO = 'SERVER_INFO',
}

export interface ServerInfoAction extends Action<ActionType.SERVER_INFO> {
  value: ServerInfo;
}

export interface Demo {
  name: string;
  visits: number;
}

export type BoundingBox = [LatLng, LatLng];

export interface ServerInfo {
  boundingBox: BoundingBox | null;
  countryCodes: string[];
  demos: Demo[];
}
