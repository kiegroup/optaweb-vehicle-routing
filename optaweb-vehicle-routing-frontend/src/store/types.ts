/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Action } from 'redux';
import { ThunkAction } from 'redux-thunk';
import WebSocketClient from 'websocket/WebSocketClient';
import { UserViewport } from './client/types';
import { Demo } from './demo/types';
import { RoutingPlan } from './route/types';
import { ServerInfo } from './server/types';
import { WebSocketConnectionStatus } from './websocket/types';

/**
 * ThunkCommand is a ThunkAction that has no result (it's typically something like
 * `Promise<ActionAfterDataFetched>`, but sending messages over WebSocket usually has no response
 * (with the exception of subscribe), so most of our operations are void).
 *
 * @template A Type of action(s) allowed to be dispatched.
 */
export type ThunkCommand<A extends Action> = ThunkAction<void, AppState, WebSocketClient, A>;

/**
 * Factory method that takes a value and creates an @type {Action}.
 *
 * @template V value type
 * @template A action type
 */
export type ActionFactory<V, A extends Action> = V extends void ?
  // https://stackoverflow.com/questions/55646272/conditional-method-parameters-based-on-generic-type
  () => A : // nullary
  (value: V) => A; // unary

/**
 * Factory method that takes a value and creates a @type {ThunkCommand}.
 *
 * @template V value type
 * @template A action type
 */
export type ThunkCommandFactory<V, A extends Action> = V extends void ?
  () => ThunkCommand<A> : // nullary
  (value: V) => ThunkCommand<A>; // unary

export interface AppState {
  readonly serverInfo: ServerInfo;
  readonly plan: RoutingPlan;
  readonly connectionStatus: WebSocketConnectionStatus;
  readonly demo: Demo;
  readonly userViewport: UserViewport;
}
