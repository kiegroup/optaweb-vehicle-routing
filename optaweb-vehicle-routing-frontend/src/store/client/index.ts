import * as clientOperations from './operations';
import { clientReducer } from './reducers';

export { clientOperations };

export default clientReducer;
