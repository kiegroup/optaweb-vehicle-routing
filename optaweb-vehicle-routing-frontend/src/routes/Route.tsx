/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {
  FormSelect,
  FormSelectOption,
  Split,
  SplitItem,
  Text,
  TextContent,
  TextVariants,
} from '@patternfly/react-core';
import LocationList from 'components/LocationList';
import TspMap from 'components/TspMap';
import * as React from 'react';
import { connect } from 'react-redux';
import { IAppState } from 'store/configStore';
import { demoOperations } from 'store/demo';
import { routeOperations, routeSelectors } from 'store/route';
import { ILatLng, IRouteWithSegments } from 'store/route/types';

export interface IStateProps {
  route: IRouteWithSegments;
  domicileId: number;
  isDemoLoading: boolean;
}

export interface IDispatchProps {
  removeHandler: typeof routeOperations.deleteLocation;
  loadHandler: typeof demoOperations.loadDemo;
  clearHandler: typeof routeOperations.clearRoute;
  addHandler: typeof routeOperations.addLocation;
}

const mapStateToProps = ({ route, demo }: IAppState): IStateProps => ({
  domicileId: routeSelectors.getDomicileId(route),
  isDemoLoading: demo.isLoading,
  route,
});

const mapDispatchToProps: IDispatchProps = {
  addHandler: routeOperations.addLocation,
  clearHandler: routeOperations.clearRoute,
  loadHandler: demoOperations.loadDemo,
  removeHandler: routeOperations.deleteLocation,
};

type IRouteProps = IDispatchProps & IStateProps;

export interface IRouteState {
  center: ILatLng;
  selectedId: number;
  zoom: number;
}

class Route extends React.Component<IRouteProps, IRouteState> {
  constructor(props: IRouteProps) {
    super(props);

    this.state = {
      center: {
        lat: 50.85,
        lng: 4.35,
      },
      selectedId: NaN,
      zoom: 9,
    };
    this.onSelectLocation = this.onSelectLocation.bind(this);
    this.handleMapClick = this.handleMapClick.bind(this);
  }

  handleMapClick(e: any) {
    this.props.addHandler(e.latlng);
  }

  onSelectLocation(id: number) {
    this.setState({ selectedId: id });
  }

  render() {
    const { center, zoom, selectedId } = this.state;
    const {
      route,
      domicileId,
      removeHandler,
      loadHandler,
      clearHandler,
      isDemoLoading,
    } = this.props;
    return (
      <React.Fragment>
        <TextContent>
          <Text component={TextVariants.h1}>Route</Text>
        </TextContent>
        <Split gutter="md">
          <SplitItem isMain={false}>
            <FormSelect
              value={''}
              // tslint:disable-next-line:no-console
              onChange={e => console.log(e)}
              aria-label="FormSelect Input"
            >
              {[{ disabled: false, value: 'wip', label: 'Vehicle 4' }].map(
                (option, index) => (
                  <FormSelectOption
                    isDisabled={option.disabled}
                    key={index}
                    value={option.value}
                    label={option.label}
                  />
                ),
              )}
            </FormSelect>
            <div
              style={{
                maxHeight: 'calc(100vh - 228px)',
                overflowY: 'auto',
              }}
            >
              <LocationList
                route={route}
                domicileId={domicileId}
                removeHandler={removeHandler}
                selectHandler={this.onSelectLocation}
                loadHandler={loadHandler}
                clearHandler={clearHandler}
                isDemoLoading={isDemoLoading}
              />
            </div>
          </SplitItem>
          <SplitItem isMain={true}>
            <TspMap
              center={center}
              zoom={zoom}
              selectedId={selectedId}
              clickHandler={this.handleMapClick}
              removeHandler={removeHandler}
              route={route}
              domicileId={domicileId}
            />
          </SplitItem>
        </Split>
      </React.Fragment>
    );
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Route);
