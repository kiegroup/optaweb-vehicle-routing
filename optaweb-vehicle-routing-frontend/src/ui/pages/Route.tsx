/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {
  Form,
  FormSelect,
  FormSelectOption,
  GutterSize,
  Split,
  SplitItem,
  Text,
  TextContent,
  TextVariants,
} from '@patternfly/react-core';
import * as React from 'react';
import { connect } from 'react-redux';
import { routeOperations } from 'store/route';
import { LatLng, Location, RouteWithTrack } from 'store/route/types';
import { AppState } from 'store/types';
import LocationList from 'ui/components/LocationList';
import TspMap from 'ui/components/TspMap';

export interface StateProps {
  depot: Location | null;
  routes: RouteWithTrack[];
  boundingBox: [LatLng, LatLng] | null;
}

export interface DispatchProps {
  addHandler: typeof routeOperations.addLocation;
  removeHandler: typeof routeOperations.deleteLocation;
}

const mapStateToProps = ({ plan, serverInfo }: AppState): StateProps => ({
  depot: plan.depot,
  routes: plan.routes,
  boundingBox: serverInfo.boundingBox,
});

const mapDispatchToProps: DispatchProps = {
  addHandler: routeOperations.addLocation,
  removeHandler: routeOperations.deleteLocation,
};

export type RouteProps = DispatchProps & StateProps;

export interface RouteState {
  selectedId: number;
  selectedRouteId: number;
}

export class Route extends React.Component<RouteProps, RouteState> {
  constructor(props: RouteProps) {
    super(props);

    this.state = {
      selectedId: NaN,
      selectedRouteId: 0,
    };
    this.onSelectLocation = this.onSelectLocation.bind(this);
    this.handleMapClick = this.handleMapClick.bind(this);
  }

  handleMapClick(e: any) {
    this.props.addHandler({ ...e.latlng, description: '' });
  }

  onSelectLocation(id: number) {
    this.setState({ selectedId: id });
  }

  render() {
    const { selectedId, selectedRouteId } = this.state;
    const {
      boundingBox,
      depot,
      routes,
      removeHandler,
    } = this.props;

    // FIXME quick hack to preserve route color by keeping its index
    const filteredRoutes = (
      routes.map((value, index) => (index === selectedRouteId ? value : { visits: [], track: [] }))
    );
    const filteredVisits: Location[] = routes.length > 0 ? routes[selectedRouteId].visits : [];
    return (
      <>
        <TextContent>
          <Text component={TextVariants.h1}>Route</Text>
        </TextContent>
        <Split gutter={GutterSize.md}>
          <SplitItem
            isFilled={false}
            style={{ display: 'flex', flexDirection: 'column' }}
          >
            <Form>
              <FormSelect
                style={{ backgroundColor: 'white', marginBottom: 10 }}
                isDisabled={routes.length === 0}
                value={selectedRouteId}
                onChange={(e) => {
                  this.setState({ selectedRouteId: parseInt(e as unknown as string, 10) });
                }}
                aria-label="FormSelect Input"
              >
                {routes.map(
                  (option, index) => (
                    <FormSelectOption
                      isDisabled={false}
                      // eslint-disable-next-line react/no-array-index-key
                      key={index}
                      value={index}
                      label={`Route ${index}`}
                    />
                  ),
                )}
              </FormSelect>
            </Form>
            <LocationList
              depot={depot}
              visits={filteredVisits}
              removeHandler={removeHandler}
              selectHandler={this.onSelectLocation}
            />
          </SplitItem>
          <SplitItem isFilled={true}>
            <TspMap
              boundingBox={boundingBox}
              selectedId={selectedId}
              clickHandler={this.handleMapClick}
              removeHandler={removeHandler}
              depot={depot}
              routes={filteredRoutes}
            />
          </SplitItem>
        </Split>
      </>
    );
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Route);
