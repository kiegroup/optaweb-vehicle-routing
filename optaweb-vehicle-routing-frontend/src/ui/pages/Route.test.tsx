/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import { UserViewport } from 'store/client/types';
import { Route, RouteProps } from './Route';

describe('Route page', () => {
  it('should render correctly with no routes', () => {
    const routes = shallow(<Route {...noRoutes} />);
    expect(toJson(routes)).toMatchSnapshot();
  });

  it('should render correctly with a few routes', () => {
    const routes = shallow(<Route {...twoRoutes} />);
    expect(toJson(routes)).toMatchSnapshot();
  });
});

const userViewport: UserViewport = {
  isDirty: false,
  zoom: 1,
  center: [0, 0],
};

const noRoutes: RouteProps = {
  addHandler: jest.fn(),
  removeHandler: jest.fn(),
  updateViewport: jest.fn(),

  boundingBox: null,
  userViewport,

  depot: null,
  visits: [],
  routes: [],
};

const depot = {
  id: 1,
  lat: 1.345678,
  lng: 1.345678,
};
const visit2 = {
  id: 2,
  lat: 2.345678,
  lng: 2.345678,
};
const visit3 = {
  id: 3,
  lat: 3.676111,
  lng: 3.568333,
};
const visit4 = {
  id: 4,
  lat: 4.345678,
  lng: 4.345678,
};
const visit5 = {
  id: 5,
  lat: 5.345678,
  lng: 5.345678,
};

const twoRoutes: RouteProps = {
  addHandler: jest.fn(),
  removeHandler: jest.fn(),
  updateViewport: jest.fn(),

  boundingBox: null,
  userViewport,

  depot,
  visits: [visit2, visit3, visit4, visit5],

  routes: [{
    vehicle: { id: 1, name: 'v1', capacity: 5 },
    visits: [depot, visit2, visit3],

    track: [],

  }, {
    vehicle: { id: 2, name: 'v2', capacity: 5 },
    visits: [depot, visit4, visit5],

    track: [],

  }],
};
