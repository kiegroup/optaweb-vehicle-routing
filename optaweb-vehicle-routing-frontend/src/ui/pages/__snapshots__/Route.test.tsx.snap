// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route page should render correctly with a few routes 1`] = `
<Fragment>
  <TextContent>
    <Text
      component="h1"
    >
      Route
    </Text>
  </TextContent>
  <Split
    className=""
    component="div"
    gutter="md"
  >
    <SplitItem
      className=""
      isFilled={false}
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <Form>
        <FormSelect
          aria-label="FormSelect Input"
          className=""
          isDisabled={false}
          isValid={true}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "marginBottom": 10,
            }
          }
          value={0}
        >
          <FormSelectOption
            className=""
            isDisabled={false}
            key="0"
            label="v1"
            value={0}
          />
          <FormSelectOption
            className=""
            isDisabled={false}
            key="1"
            label="v2"
            value={1}
          />
        </FormSelect>
      </Form>
      <LocationList
        depot={
          Object {
            "id": 1,
            "lat": 1.345678,
            "lng": 1.345678,
          }
        }
        removeHandler={[MockFunction]}
        selectHandler={[Function]}
        visits={
          Array [
            Object {
              "id": 1,
              "lat": 1.345678,
              "lng": 1.345678,
            },
            Object {
              "id": 2,
              "lat": 2.345678,
              "lng": 2.345678,
            },
            Object {
              "id": 3,
              "lat": 3.676111,
              "lng": 3.568333,
            },
          ]
        }
      />
    </SplitItem>
    <SplitItem
      className=""
      isFilled={true}
    >
      <RouteMap
        boundingBox={null}
        clickHandler={[Function]}
        depot={
          Object {
            "id": 1,
            "lat": 1.345678,
            "lng": 1.345678,
          }
        }
        removeHandler={[MockFunction]}
        routes={
          Array [
            Object {
              "track": Array [],
              "vehicle": Object {
                "capacity": 5,
                "id": 1,
                "name": "v1",
              },
              "visits": Array [
                Object {
                  "id": 1,
                  "lat": 1.345678,
                  "lng": 1.345678,
                },
                Object {
                  "id": 2,
                  "lat": 2.345678,
                  "lng": 2.345678,
                },
                Object {
                  "id": 3,
                  "lat": 3.676111,
                  "lng": 3.568333,
                },
              ],
            },
            Object {
              "track": Array [],
              "visits": Array [],
            },
          ]
        }
        selectedId={NaN}
        updateViewport={[MockFunction]}
        userViewport={
          Object {
            "center": Array [
              0,
              0,
            ],
            "isDirty": false,
            "zoom": 1,
          }
        }
        visits={
          Array [
            Object {
              "id": 2,
              "lat": 2.345678,
              "lng": 2.345678,
            },
            Object {
              "id": 3,
              "lat": 3.676111,
              "lng": 3.568333,
            },
            Object {
              "id": 4,
              "lat": 4.345678,
              "lng": 4.345678,
            },
            Object {
              "id": 5,
              "lat": 5.345678,
              "lng": 5.345678,
            },
          ]
        }
      />
    </SplitItem>
  </Split>
</Fragment>
`;

exports[`Route page should render correctly with no routes 1`] = `
<Fragment>
  <TextContent>
    <Text
      component="h1"
    >
      Route
    </Text>
  </TextContent>
  <Split
    className=""
    component="div"
    gutter="md"
  >
    <SplitItem
      className=""
      isFilled={false}
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <Form>
        <FormSelect
          aria-label="FormSelect Input"
          className=""
          isDisabled={true}
          isValid={true}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "marginBottom": 10,
            }
          }
          value={0}
        />
      </Form>
      <LocationList
        depot={null}
        removeHandler={[MockFunction]}
        selectHandler={[Function]}
        visits={Array []}
      />
    </SplitItem>
    <SplitItem
      className=""
      isFilled={true}
    >
      <RouteMap
        boundingBox={null}
        clickHandler={[Function]}
        depot={null}
        removeHandler={[MockFunction]}
        routes={Array []}
        selectedId={NaN}
        updateViewport={[MockFunction]}
        userViewport={
          Object {
            "center": Array [
              0,
              0,
            ],
            "isDirty": false,
            "zoom": 1,
          }
        }
        visits={Array []}
      />
    </SplitItem>
  </Split>
</Fragment>
`;
